sourcegen-bytecode-writer/src/main/java/io/micronaut/sourcegen/bytecode/expression/VariableExpressionWriter.java

```diff
@@ -82,4 +77,56 @@ public void write(GeneratorAdapter generatorAdapter, MethodContext context) {
         }
         throw new UnsupportedOperationException("Unrecognized variable: " + variableDef);
     }
+
+    private void writeField(GeneratorAdapter generatorAdapter, MethodContext context, VariableDef.Field field) {
+        ExpressionWriter.writeExpression(generatorAdapter, context, field.instance());
+        TypeDef fieldType = field.type();
+        TypeDef owner = field.instance().type();
+        generatorAdapter.getField(TypeUtils.getType(owner, context.objectDef()), field.name(), TypeUtils.getType(fieldType, context.objectDef()));
+    }
+
+    private void writeParameterVariable(GeneratorAdapter generatorAdapter, MethodContext context, String name) {
+        if (context.methodDef() == null) {
+            throw new IllegalStateException("Accessing method parameters is not available");
+        }
+        ParameterDef parameterDef = context.methodDef().getParameters().stream().filter(p -> p.getName().equals(name)).findFirst().orElseThrow();
+        int parameterIndex = context.methodDef().getParameters().indexOf(parameterDef);
+        generatorAdapter.loadArg(parameterIndex);
+    }
+
+    private void writeStaticField(GeneratorAdapter generatorAdapter, MethodContext context, VariableDef.StaticField field) {
+        TypeDef owner = field.ownerType();
+        TypeDef fieldType = field.type();
+
+        generatorAdapter.getStatic(TypeUtils.getType(owner, context.objectDef()), field.name(), TypeUtils.getType(fieldType, context.objectDef()));
+    }
+
+    public void writeLambdaVariable(GeneratorAdapter generatorAdapter, MethodContext context) {
+        if (variableDef instanceof VariableDef.StaticField field) {
+            writeStaticField(generatorAdapter, context, field);
+            return;
+        } else if (variableDef instanceof VariableDef.Field field) {
+            writeField(generatorAdapter, context, field);
+            return;
+        }
+        String name = null;
+        if (variableDef instanceof VariableDef.ExceptionVar) {
+            name = LambdaExpressionWriter.EXCEPTION_VAR_NAME;
+        } else if (variableDef instanceof VariableDef.This) {
+            name = LambdaExpressionWriter.THIS_VAR_NAME;
+        } else if (variableDef instanceof VariableDef.Super) {
+            name = LambdaExpressionWriter.SUPER_VAR_NAME;
+        } else if (variableDef instanceof VariableDef.Local local) {
+            name = local.name();
+        } else if (variableDef instanceof VariableDef.MethodParameter methodParameter) {
+            name = methodParameter.name();
+        }
+
+        if (name != null) {
+            writeParameterVariable(generatorAdapter, context, name);
+            return;
+        }
+        throw new UnsupportedOperationException("Unrecognized lambda variable: " + variableDef);
```

Comment:
the processingexception instead

Reply:
There isn't always a visitor context available (the classes are not always generated during annotation processing).